File: 1725b_bt.cpp
/**
 * @file 1725b_bt.cpp
 * @author Ukeme Edet (ukemeedet2207@gmail.com)
 * @brief The solution to the problem Basketball Together(http://codeforces.com/contest/1725/problem/B)
 * @version 0.1
 * @date 2024-03-28
 *
 * @copyright Copyright (c) 2024
 *
 */

#include <bits/stdc++.h>

using namespace std;

/**
 * @brief The main function
 *
 * @return int
 */
int main(void)
{
	ios::sync_with_stdio(0);
	cin.tie(0);
	int n, d, i, ts, nc, wins;
	vector<int> p;

	cin >> n >> d;
	p.resize(n);
	for (int &pi : p)
		cin >> pi;
	sort(p.rbegin(), p.rend());
	wins = 0, nc = n;
	for (i = 0; i < n; i++, wins++)
	{
		ts = ceil((double)d / p[i]);
		!(d % p[i]) ? ts++ : ts;
		if (nc - ts < 0)
			break;
		nc -= ts;
	}
	cout << wins << '\n';
	return (EXIT_SUCCESS);
}

File: 1725b_bt.py
"""
@file: 1725b_bt.py
@author: Ukeme Edet (ukemeedet2207@gmail.com)
@brief: The solution to the problem Basketball Together(https://codeforces.com/problemset/problem/1725/B)
@version: 1.0
@date: 2024-11-17

@copyright: Copyright (c) 2024
"""

from sys import stdin, stdout


def input():
    return stdin.readline().strip()


def print(string):
    return stdout.write(str(string) + "\n")


def main():
    n, d = [int(x) for x in input().split()]
    p = sorted([int(x) for x in input().split()])
    res = cp = i = 0
    while n:
        cp += p[-i - 1]
        n -= 1
        if cp > d:
            cp = 0
            i += 1
            res += 1
    print(res)


if __name__ == "__main__":
    main()
